gcp/gke.ts

```diff
@@ -0,0 +1,115 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as k8s from "@pulumi/kubernetes";
+import * as gcp from "@pulumi/gcp";
+
+import * as model from "./model";
+import * as config from "./config";
+
+class GcpKubernetesCluster implements model.KubernetesCluster {
+  cluster: gcp.container.Cluster;
+  kubeconfig: pulumi.Output<any>;
+  name: pulumi.Output<any>;
+  k8sProvider: k8s.Provider;
+
+  constructor(cluster: gcp.container.Cluster, kubeconfig: pulumi.Output<any>, provider: k8s.Provider) {
+    this.cluster = cluster;
+    this.kubeconfig = kubeconfig;
+    this.name = cluster.id;
+    this.k8sProvider = provider;
+  }
+}
+
+export class GcpCloud implements model.Cloud {
+  /**
+   * Creates a GCP cluster.
+   */
+   createKubernetesCluster(): model.KubernetesCluster {
+     // Create a GKE cluster
+     const engineVersion = gcp.container.getEngineVersions().then(v => v.latestMasterVersion);
+     const cluster = new gcp.container.Cluster(config.clusterName, {
+       location: config.zone,
+       initialNodeCount: 1,
+       minMasterVersion: engineVersion,
+       nodeVersion: engineVersion,
+       removeDefaultNodePool: true
+     });
+
+     const primaryPreemptibleNodes = new gcp.container.NodePool("primarynodes", {
+       location: config.zone,
+       cluster: cluster.name,
+       initialNodeCount: config.initialNodeCountInCluster,
+       autoscaling: {
+         maxNodeCount: config.autoscalingMaxNodeCount,
+         minNodeCount: config.autoscalingMinNodeCount
+       },
+       nodeConfig: {
+         preemptible: true,
+         machineType: "n1-standard-4",
```

Comment:
Can this also be a variable (with default)?

Reply:
👍